server/svix-server/src/queue/redis.rs

```diff
@@ -56,61 +74,131 @@ fn regenerate_key(msg: &str) -> String {
     to_redis_key(&delivery)
 }
 
-pub async fn new_pair(pool: RedisPool) -> (TaskQueueProducer, TaskQueueConsumer) {
-    let worker_pool = pool.clone();
-    tokio::spawn(async move {
-        // FIXME: enforce we only have one such worker (locking)
-        let batch_size: isize = 50;
-        let task_validity_duration = chrono::Duration::from_std(TASK_VALIDITY_DURATION).unwrap();
+pub async fn new_pair(
+    pool: RedisPool,
+    pending_duration: Duration,
+) -> (TaskQueueProducer, TaskQueueConsumer) {
+    // Create the stream and consumer group for the MAIN queue should it not already exist
+    {
+        let mut conn = pool
+            .get()
+            .await
+            .expect("Error retreiving connection from Redis pool");
 
-        {
-            let pool = worker_pool.clone();
```

Comment:
Make the diff smaller. If you want to make this change, do it later, in a different PR.